mlflow/pyfunc/__init__.py

```diff
@@ -1052,6 +1053,36 @@ def _get_pip_requirements_from_model_path(model_path: str):
 
     return [req.req_str for req in _parse_requirements(req_file_path, is_constraint=False)]
 
+class _LoadedModelTracker:
```

Comment:
@prithvikannan Can we add test coverage for pyfunc, langchain (if not already present), and one other model flavor?

Reply:
cc @harupy @serena-ruan in case it's better for y'all to own the test coverage

Reply:
we have test coverage for LC, but yea ill add for pyfunc